com.unity.shadergraph/Tests/Registry/RegistryPlaceholderTest.cs

```diff
@@ -5,63 +5,65 @@
+        //    public void RegistryPlaceholderTest()
+        //    {
+        //        var graph = new GraphHandler();
+        //        var registry = new Registry();
+
+        //        registry.Register<GraphType>();
+        //        _ = registry.Register<TestAddNode>();
+        //        registry.Register<GraphTypeAssignment>();
+
+        //        // should default concretize length to 4.
+        //        graph.AddNode<TestAddNode>("Add1", registry);
+        //        var reader = graph.GetNodeReader("Add1");
+        //        reader.TryGetField("In1.TypeField.Length", out GraphType.Length len);
+        //        Assert.AreEqual(4, (int)len);
+
+        //        // Set the length of input port 1 to 1.
+        //        var nodeWriter = graph.GetNodeWriter("Add1");
+        //        nodeWriter.SetPortField("In1", "Length", GraphType.Length.One);
+
+        //        // After reconcretization, the node definition should propagate the length.
+        //        graph.ReconcretizeNode("Add1", registry);
+        //        reader = graph.GetNodeReader("Add1");
+        //        reader.TryGetField("In1.TypeField.Length", out len);
+        //        Assert.AreEqual(1, (int)len);
+        //        reader.TryGetField("In2.TypeField.Length", out len);
+        //        Assert.AreEqual(1, (int)len);
+        //        reader.TryGetField("Out.TypeField.Length", out len);
+        //        Assert.AreEqual(1, (int)len);
+
+        //        // Add a second Add Node, with length 2 this time.
+        //        var node2 = graph.AddNode<TestAddNode>("Add2", registry);
+        //        node2.SetPortField("In2", "Length", GraphType.Length.Two);
+        //        graph.ReconcretizeNode("Add2", registry);
+        //        reader = graph.GetNodeReader("Add2");
+        //        reader.TryGetField("In1.TypeField.Length", out len);
+        //        Assert.AreEqual(2, (int)len);
+        //        reader.TryGetField("In2.TypeField.Length", out len);
+        //        Assert.AreEqual(2, (int)len);
+        //        reader.TryGetField("Out.TypeField.Length", out len);
+        //        Assert.AreEqual(2, (int)len);
+
+        //        // Connecting Out to In should clobber the inlined length with the new length.
+        //        graph.TryConnect("Add2", "Out", "Add1", "In1", registry);
+        //        graph.ReconcretizeNode("Add1", registry);
+        //        reader = graph.GetNodeReader("Add1");
+        //        reader.TryGetPort("In1", out var portReader);
+        //        portReader.GetTypeField().GetField("Length", out len);
+        //        Assert.AreEqual(2, (int)len);
+        //        reader.TryGetField("In2.TypeField.Length", out len);
+        //        Assert.AreEqual(2, (int)len);
+        //        reader.TryGetField("Out.TypeField.Length", out len);
+        //        Assert.AreEqual(2, (int)len);
+        //    }
```

Comment:
Concretization changes mean that this is no longer expected to work, but I am leaving it in comments for now. We may update TestAddNode or we may replace it later.